lib/api/core/entrypoints/index.js

```diff
@@ -19,51 +19,529 @@
+            maxFiles: conf.maxFiles,
+            eol: conf.eol || '\n',
+            logstash: conf.logstash || false,
+            tailable: conf.tailable,
+            maxRetries: conf.maxRetries || 2,
+            zippedArchive: conf.zippedArchive || false
+          })));
+          break;
+        case 'syslog':
+          transports.push(new WinstonSyslog(Object.assign(opts, {
+            host: conf.host || 'localhost',
+            port: conf.port || 514,
+            protocol: conf.protocol || 'udp4',
+            path: conf.path || '/dev/log',
+            pid: conf.pid || process.pid,
+            facility: conf.facility || 'local0',
+            localhost: conf.localhost || 'localhost',
+            type: conf.type || 'BSD',
+            app_name: conf.app_name || process.title,
+            eol: conf.eol
+          })));
+          break;
+        default:
+          this.kuzzle.log.error(
+            `Failed to initialize logger transport "${conf.transport}": unsupported transport. Skipped.`);
+      }
+    }
+
+    this.logger = winston.createLogger({transports});
+  }
+
+  /**
+   * /!\ This method is a critical section of code.
+   * Modifications must be compared against current
+   * performances benchmarks
+   *
+   * @param {Request} request
+   * @param {object} extra
+   */
+  logAccess (request, extra = null) {
+    let connection = this.clients[request.context.connection.id];
+
+    // Make do with the RequestContext information
+    if (!connection) {
+      connection = new ClientConnection(
+        request.context.connection.protocol || '-',
+        request.context.connection.ips || [],
+        request.context.connection.misc.headers
+      );
+    }
+
+    if (this.config.logs.accessLogFormat === 'logstash') {
+      // custom kuzzle logs to be exported to logstash
+      this.logger.info({
+        connection,
+        extra,
+        request: request.input,
+        error: request.error,
+        status: request.status
+      });
+      return;
+    }
+
+    // user init: prioritize the already decoded and verified token stored in
+    // the request
+    // If not available, then that may mean that we didn't verify the user yet,
+    // so we have to decode any provided token
+    let user = null;
+
+    if (request.context.token !== null) {
+      if (request.context.token.userId === this.anonymousUserId) {
+        user = '(anonymous)';
+      } else {
+        user = request.context.token.userId;
+      }
+    }
+
+    // = apache combined
+    const protocol = connection.protocol.toUpperCase();
+    let
+      url,
+      verb = 'DO';
+
+    if (connection.protocol.startsWith('HTTP/')) {
+      verb = extra.method;
+      url = extra.url;
+
+      // try to get plain user name, 1st form jwt token if present, then from
+      // basic auth
+      if (user === null && connection.headers.authorization) {
+        try {
+          if (bearerRegexp.test(connection.headers.authorization)) {
```

Comment:
Won't work because this can also be written `BeAREr`

But all you need is an indexOf on a lowercased string, which gives similar results to yours. :+1: